GSoC-2025/ethology_support_for_any_point_tracking_vietluong.md

```diff
@@ -0,0 +1,148 @@
+
+
+## ethology: Support for Any-Point Tracking (Viet Luong)
+
+## Personal details
+Please include the following information:
+- **Full name**: Viet Luong
+- **Email**: vietintern@gmail.com
+- **GitHub username**:vietluong2110
+- **Zulip username**: Viet Luong
+- **Location & time-zone**: Chicago, GMT-5
+- **Personal website / project portfolio**
+- **Code contribution**
+
+    - [add tapir option to tap models api #73 - In Review](https://github.com/neuroinformatics-unit/ethology/pull/73)
+    - [video-utilities - In Review](https://github.com/neuroinformatics-unit/ethology/pull/68)
+    - [JAX-SimpleTM](https://github.com/vietluong2110/jax_simpletm)    
+
+- **Proposal discussion link**
+
+    Please link to the pull request where you discussed your project proposal with the community. 
+
+## Project proposal 
+_Length: max 1 page_
+
+- **Synopsis**
+
+This project integrates advanced any-point tracking methods like cotracker3 and TAPIR into the ethology framework, aiding animal behavior research. It will develop a napari plugin for video loading, point selection, automated tracking, and trajectory visualization. Deliverables include a napari widget prototype, back-end tracking integration, and front-end visualization. The open-source community benefits from improved accessibility to advanced computer vision tools, enhancing research efficiency.
+
+- **Implementation timeline**
+
+**Minimal Set of Deliverables:**
+- Functional napari widget prototype for loading videos and selecting query points.
+- Back-end integration with cotracker3 or TAPIR for running any-point tracking inference.
+- Capability to generate and export trajectory data.
+- Visualization overlay of trajectories within the napari interface.
+- Documentation for plugin usage and codebase.
+
+ **Stretch Goals (If time allows)**:
+- Real-time inference directly within the napari GUI.
+- Integration of additional any-point tracking models.
+- Advanced visualization features (trajectory filtering, custom annotations).
+
+**Community Bonding Period (Prep work)**:
+- Familiarize thoroughly with napari plugin documentation and development process.
+- Set up the development environment.
+- Discuss implementation details and milestones with mentors.
+
+**Weekly Timeline**
```

Comment:
I think this timeline is quite front-loaded, meaning a big chunk of the work is assigned (maybe unnecessarily?) to weeks 1-6. I would maybe revisit it, and try to spread the work more uniformly across the whole period. For this it may help to think about how would you split the tasks into the smallest subtasks possible, and about the specific steps you would need to take to implement the desired features.